packages/mds-compliance/api.ts

```diff
@@ -159,13 +159,20 @@ function api(app: express.Express): express.Express {
   })
 
   app.get(pathsFor('/count/:rule_id'), async (req: ComplianceApiRequest, res: ComplianceApiResponse) => {
+    // /*
+    const { timestamp: query_end_date } = {
+      ...parseRequest(req, { parser: Number }).query('timestamp')
+    }
+    //* /
     if (!AllowedProviderIDs.includes(res.locals.provider_id)) {
       return res.status(401).send({ result: 'unauthorized access' })
     }
 
+    const timestamp = query_end_date || now()
+
     async function fail(err: Error) {
       logger.error(err.stack || err)
-      if (err.message.includes('invalid rule_id')) {
+      if (err.message.includes('invalid rule_id') || err.message.includes('not found in any policy')) {
```

Comment:
I think we should try to avoid adding more things into this error handling pattern if possible

Reply:
+1, if we need to do this sort of error handling branching I'd rather be comparing status codes, string enums, or doing `instanceof` checks.

Reply:
Fixing the error handling feels out of scope. I'd rather keep this PR focused on handling the `timestamp`param.

Reply:
Sounds reasonable, maybe a good candidate for a follow up PR

Reply:
I was going to split that out into a different PR so it'd be easier to read, but I think it's faster to just stick it all in this one.